ClientApp/viking-vault/src/UserDashboard/TransactionList.tsx

```diff
@@ -4,15 +4,18 @@ import "./TransactionList.css";
 import shoppingCart from '../Resources/images/shoppingcart.png';
 import userIcon from '../Resources/images/profileWhite.png';
 import exchange from "../Resources/images/money-exchange.png";
+import { IUserData } from './../AdminDashboard/UserData';
 
 const baseUrl = constants.baseUrl;
 
 interface ITransaction {
+    user: IUserData;
     type: string;
     date: Date;
     amount: number;
     currency: string;
-    otherParty: string;
+    senderOrReceiver: IUserData;
```

Comment:
Ever heard of SRP? :)

Reply:
No 😒

Reply:
> Ever heard of SRP? :)

SenderReceiverParty?

Reply:
senderOrReceiver has 2 purposes and it is not a good practice. Please rewrite your code or rename it such that it only has one purpose to exist.